sql/atxs/atxs.go

```diff
@@ -73,6 +72,29 @@ func GetTimestamp(db sql.Executor, id types.ATXID) (timestamp time.Time, err err
 	return timestamp, err
 }
 
+// GetFirstIDByNodeID gets the initial ATX ID for a given node ID.
+func GetFirstIDByNodeID(db sql.Executor, nodeID types.NodeID) (id types.ATXID, err error) {
+	enc := func(stmt *sql.Statement) {
+		stmt.BindBytes(1, nodeID.ToBytes())
+	}
+	dec := func(stmt *sql.Statement) bool {
+		stmt.ColumnBytes(0, id[:])
+		return true
+	}
+
+	if rows, err := db.Exec(`
+		select id from atxs 
+		where smesher = ?1
+		order by epoch asc, timestamp asc
+		limit 1;`, enc, dec); err != nil {
+		return types.ATXID{}, fmt.Errorf("exec id %v: %w", id, err)
```

Comment:
this error string does not mke sense. id is always empty in this case. i think you want `nodeID` instead. actually this is the same for other methods as well.

Reply:
Fixed, also for the other methods